src/AspNetCore/Src/applicationinfo.cpp

```diff
@@ -146,12 +155,22 @@ APPLICATION_INFO::UpdateAppOfflineFileHandle()
                     strEventMsg.QueryStr());
             }
 
-            m_pApplication->ShutDown();
-            m_pApplication->DereferenceApplication();
-            m_pApplication = NULL;
+            APPLICATION_MANAGER::GetInstance()->RecycleApplicationFromManager(
```

Comment:
you should call recycleapplication here which will just recycle specific application instance (as this is for app_offline) not for configuration change. Otherwise a subapplication may get recycled